File: autocomplete/frontend/src/types/product.ts
// src/types/product.ts

/**
 * Product TypeScript Interfaces
 * 
 * These interfaces define the expected structure for product data
 * as returned by the DummyJSON API, per assignment requirements and
 * API documentation: https://dummyjson.com/docs/products
 * 
 * Each product has:
 * - id
 * - title
 * - brand
 * - category
 * - price
 * 
 * The API response for a search includes:
 * - products: Product[]
 * - total: number
 * - skip: number
 * - limit: number
 */

export interface Product {
  id: number;
  title: string;
  brand: string;
  category: string;
  price: number;
  // Optionally, you can add:
  // description?: string;
  // thumbnail?: string;
  // images?: string[];
}

export interface ProductSearchResponse {
  products: Product[];
  total: number;
  skip: number;
  limit: number;
}

File: backend/src/models/product.py
# backend/src/models/product.py

from pydantic import BaseModel, Field

class Product(BaseModel):
    id: int = Field(..., example=1)
    title: str = Field(..., example="iPhone 13")
    brand: str = Field(..., example="Apple")
    category: str = Field(..., example="smartphones")
    price: float = Field(..., example=799.99)

    class Config:
        schema_extra = {
            "example": {
                "id": 1,
                "title": "iPhone 13",
                "brand": "Apple",
                "category": "smartphones",
                "price": 799.99
            }
        }

class ProductSearchResponse(BaseModel):
    products: list[Product]
    total: int
    skip: int
    limit: int

    class Config:
        schema_extra = {
            "example": {
                "products": [
                    {
                        "id": 1,
                        "title": "iPhone 13",
                        "brand": "Apple",
                        "category": "smartphones",
                        "price": 799.99
                    }
                ],
                "total": 100,
                "skip": 0,
                "limit": 10
            }
        }
